odxtools/database.py

```diff
@@ -39,20 +37,23 @@ def __init__(self,
         self._comparam_subsets = NamedItemList[ComparamSubset]()
         self._comparam_specs = NamedItemList[ComparamSpec]()
 
-    def add_pdx_file(self, pdx_file_name: str) -> None:
-        pdx_zip = ZipFile(pdx_file_name)
-
-        for zip_member in pdx_zip.namelist():
+    def add_pdx_file(self, pdx_file: Union[str, PathLike[str], IO[bytes], ZipFile]) -> None:
+        """Add PDX file to database.
+        Either pass the path to the file, an IO with the file content or a ZipFile object.
+        """
+        if not isinstance(pdx_file, ZipFile):
+            pdx_file = ZipFile(pdx_file)
+        for zip_member in pdx_file.namelist():
             # The name of ODX files can end with .odx, .odx-d,
             # .odx-c, .odx-cs, .odx-e, .odx-f, .odx-fd, .odx-m,
             # .odx-v .  We could test for all that, or just make
             # sure that the file's suffix starts with .odx
             p = Path(zip_member)
             if p.suffix.lower().startswith(".odx"):
-                root = ElementTree.parse(pdx_zip.open(zip_member)).getroot()
+                root = ElementTree.parse(pdx_file.open(zip_member)).getroot()
                 self._process_xml_tree(root)
             elif p.name.lower() != "index.xml":
-                self.add_auxiliary_file(zip_member, pdx_zip.open(zip_member))
+                self.add_auxiliary_file(zip_member, pdx_file.open(zip_member))
 
     def add_odx_file(self, odx_file_name: str) -> None:
```

Comment:
```suggestion
    def add_odx_file(self, odx_file_name: Union[PathLike, str]) -> None:
```

Reply:
same for `add_aux_file()`  (for some reason, github does allow me to place a comment there.) line 66 should become `self.auxiliary_files[str(aux_file_name)] = aux_file_obj` then.